Fixed Windows build system.

jtesta opened the pull request:
Cross-compiling a Windows executable from Linux is now possible again.  The build system automatically pulls and compiles the OpenSSL sources from GitHub just like the UNIX build system does too.

rbsec commented:
Thanks for this - I just tried cross-compiling from a Kali system and it's working again. I never use the Windows build of sslscan, so I've not tried to do this in ages.

The build doesn't do the compression checks (no zlib support), but it looks like that would have to be downloaded and compiled as well to get it working which is a bit of a pain. It's a minor check though, so it can go on the todo list.

Would it possible to change this PR to build against PeterMosmans' fork of OpenSSL, to provide the SSLv2/EXPORT ciphers along with the new ChaCha ciphers? This change was made recently to sslscan in ef6390d025de11ea743fe4400b56de89befb1695.

Other than that it looks good to merge.

jtesta commented:
Ok.  I added zlib support.  I also switched the build to use PeterMosman's fork.

Are you using this alternate fork as a temporary fix, or a permanent one?  I think someone else suggested that we include a shim in our own project to test for SSLv2.  I like that idea better, personally, as a long-term fix.

I imagine ChaCha20 support should land in the mainline distro eventually, too, no?

rbsec commented:
Sorry, only just got round to  looking at this again. I've had a play with building it on a few different systems and it's working perfectly, so I've merged it in.

With regards to OpenSSL, last time I looked there didn't seem to be any plans to officially backport ChaCha to 1.0.2 (in the main branch anyway), so I think it'll stay on this fork for the forseeable future. The fork also re-enables some other bad things (like very weak DH keys), which were previously re-enabled with a hacky sed command (and breaking one of the tests). Unless we run into any problems with this branch I think it'll stay for a while, but I've not really thought much about the long term future of sslscan (especially once TLSv1.3 starts to become important).
